tests/ops/op_math/test_exp.py

```diff
@@ -317,10 +317,18 @@ def test_pow(self):
         assert pow_op.base is base
         assert pow_op.coeff == coeff * z
 
-    def test_label(self):
-        """Test that the label is always EXP"""
-        op = Exp(qml.PauliZ(0), 2 + 3j)
-        assert op.label(decimals=4) == "Exp"
+    @pytest.mark.parametrize(
+        "op,decimals,expected",
+        [
+            (Exp(qml.PauliZ(0), 2 + 3j), None, "Exp((2+3j) Z)"),
```

Comment:
I would also prefer to also have a test for exponentiating tensor of Pauli/Rotation ops.

Reply:
added some cases for product of Pauli ops (I did prod instead of Tensor because we're planning to replace Tensor with prod)